### PR TITLE
Aug k8s patch update

### PR DESCRIPTION
**Pull request by mitulshah-suse**
Related to https://github.com/rancher/rancher/issues/38361
Also, https://github.com/rancher/rancher/issues/38662

Earlier PR got closed by mistake https://github.com/rancher/kontainer-driver-metadata/pull/944 so opened a new one.

### CODE REVIEW

File: rke/k8s_rke_system_images.go [hunk 1]
```diff
@@ -7670,6 +7670,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			KubeDNSAutoscaler:         "rancher/mirrored-cluster-proportional-autoscaler:1.8.3",
+			Flannel:                   "rancher/mirrored-coreos-flannel:v0.15.1",
+			FlannelCNI:                "rancher/flannel-cni:v0.3.0-rancher6",
+			CalicoNode:                "rancher/mirrored-calico-node:v3.21.1",
+			CalicoCNI:                 "rancher/calico-cni:v3.21.3-rancher1",
+			CalicoControllers:         "rancher/mirrored-calico-kube-controllers:v3.21.1",
+			CalicoCtl:                 "rancher/mirrored-calico-ctl:v3.21.1",
+			CalicoFlexVol:             "rancher/mirrored-calico-pod2daemon-flexvol:v3.21.1",
+			CanalNode:                 "rancher/mirrored-calico-node:v3.21.1",
+			CanalCNI:                  "rancher/calico-cni:v3.21.1-rancher1",
```
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: This needs to be updated to `rancher/calico-cni:v3.21.3-rancher1` as well.

File: rke/k8s_rke_system_images.go [hunk 2]
```diff
@@ -7670,6 +7670,50 @@ func loadK8sRKESystemImages() map[string]v3.RKESystemImages {
...
+			PodInfraContainer:         "rancher/mirrored-pause:3.6",
+			Ingress:                   "rancher/nginx-ingress-controller:nginx-1.2.1-rancher1",
+			IngressBackend:            "rancher/mirrored-nginx-ingress-controller-defaultbackend:1.5-rancher1",
+			IngressWebhook:            "rancher/mirrored-ingress-nginx-kube-webhook-certgen:v1.1.1",
+			MetricsServer:             "rancher/mirrored-metrics-server:v0.5.1",
+			CoreDNS:                   "rancher/mirrored-coredns-coredns:1.8.6",
+			CoreDNSAutoscaler:         "rancher/mirrored-cluster-proportional-autoscaler:1.8.5",
+			WindowsPodInfraContainer:  "rancher/mirrored-pause:3.6",
+			Nodelocal:                 "rancher/mirrored-k8s-dns-node-cache:1.21.1",
+		},
```
Comment (snasovich) [inline rke/k8s_rke_system_images.go]: Nit, but there are a couple of extra trailing tabs in this line (as well as at the end of the last line for all added blocks in this PR).

**Review by snasovich — CHANGES_REQUESTED** [review]

File: rke/k8s_rke_system_images.go (hunk 1 above)
Comment (mitulshah-suse) [reply rke/k8s_rke_system_images.go]: Updated

File: rke/k8s_rke_system_images.go (hunk 2 above)
Comment (mitulshah-suse) [reply rke/k8s_rke_system_images.go]: removed the extra spaces

File: rke/k8s_version_info.go [hunk 3]
```diff
@@ -664,6 +664,10 @@ func loadK8sVersionInfo() map[string]v3.K8sVersionInfo {
 			MinRancherVersion: "2.6.3-patch0",
 			MinRKEVersion:     "1.3.3-rc0",
 		},
+		"v1.22.13-rancher1-1": {
+			MinRancherVersion: "2.6.3-patch0",
+			MinRKEVersion:     "1.3.3-rc0",
+		},
```
Comment (snasovich) [inline rke/k8s_version_info.go]: Another tiny nit, but the extra trailing tabs are not addressed here.

**Review by snasovich — CHANGES_REQUESTED** [review]
There is a tiny nit to address.

Note that there are currently 7 commits and we don't want all of them on the target branch. You have couple of options after addressing this tiny nit:
1. Squash during merge (GH provides this option)
2. Squash commits locally and force-push. Ideally we want to have 2 commits - one for all the material changes, the other one for `go generate` - e.g. https://github.com/rancher/kontainer-driver-metadata/pull/927/commits

**Comment by mitulshah-suse** [comment]
> There is a tiny nit to address.
> 
> Note that there are currently 7 commits and we don't want all of them on the target branch. You have couple of options after addressing this tiny nit:
> 
> 1. Squash during merge (GH provides this option)
> 2. Squash commits locally and force-push. Ideally we want to have 2 commits - one for all the material changes, the other one for `go generate` - e.g. https://github.com/rancher/kontainer-driver-metadata/pull/927/commits

I was following this example for the commits, so that it is easier to review what exactly changed as part of the patches. https://github.com/rancher/kontainer-driver-metadata/pull/884/commits 
Per my understanding, this is how it was requested to be raised. For now I will squash the commits during merge, if that is alright.

**Review by snasovich — APPROVED** [review]
> > There is a tiny nit to address.
> > Note that there are currently 7 commits and we don't want all of them on the target branch. You have couple of options after addressing this tiny nit:
> > 
> > 1. Squash during merge (GH provides this option)
> > 2. Squash commits locally and force-push. Ideally we want to have 2 commits - one for all the material changes, the other one for `go generate` - e.g. [#927 (commits)](https://github.com/rancher/kontainer-driver-metadata/pull/927/commits)
> 
> I was following this example for the commits, so that it is easier to review what exactly changed as part of the patches. [#884 (commits)](https://github.com/rancher/kontainer-driver-metadata/pull/884/commits) Per my understanding, this is how it was requested to be raised. For now I will squash the commits during merge, if that is alright.

Yes, I agree that pushing changes as commits makes it easier to review. Please squash the commits at merge time - we don't always follow this practice, but it makes target branch commit history more manageable.

Otherwise, LGTM.

**Pull request merged**